src/OrchardCore/OrchardCore.Configuration.KeyVault/Extensions/AzureKeyVaultConfigurationExtension.cs

```diff
@@ -74,9 +74,15 @@ private static void AddOrchardCoreAzureKeyVault(
             throw new Exception("The 'KeyVaultName' property is no configured. Please configure it by specifying the 'OrchardCore:OrchardCore_KeyVault_Azure:KeyVaultName' settings key.");
         }
 
-        if (!Uri.TryCreate($"https://{keyVaultName}.vault.azure.net", UriKind.Absolute, out var keyVaultEndpointUri))
+        var vaultURI = configuration["OrchardCore:OrchardCore_KeyVault_Azure:VaultURI"];
+
+        Uri keyVaultEndpointUri;
+
+        if ((!string.IsNullOrWhiteSpace(vaultURI) && !Uri.TryCreate(vaultURI, UriKind.Absolute, out keyVaultEndpointUri)) || 
```

Comment:
I think there is a logical error here and does not meet the comments in the app settings. I think if you provide the following settings, you'll get an exception:

```
"OrchardCore_KeyVault_Azure": {
    "KeyVaultName": "* << invalid name"
    "VaultURI": "https://abc.vault.azure.net"
}
```

I think you logic should look more like this:

```csharp
var vaultUri = configuration["OrchardCore:OrchardCore_KeyVault_Azure:VaultURI"];

Uri keyVaultEndpointUri;

if (!string.IsNullOrWhiteSpace(vaultUri) && !Uri.TryCreate(vaultUri, UriKind.Absolute, out keyVaultEndpointUri))
{
    throw new Exception("Invalid value used for 'VaultURI' property. Please provide a valid key-vault name using the 'OrchardCore:OrchardCore_KeyVault_Azure:KeyVaultName' settings key.");
} 
else 
{
    var keyVaultName = configuration["OrchardCore:OrchardCore_KeyVault_Azure:KeyVaultName"];

    if (string.IsNullOrEmpty(keyVaultName))
    {
        throw new Exception("The 'KeyVaultName' property is no configured. Please configure it by specifying the 'OrchardCore:OrchardCore_KeyVault_Azure:KeyVaultName' settings key.");
    }

    if (!Uri.TryCreate($"https://{keyVaultName}.vault.azure.net", UriKind.Absolute, out var keyVaultEndpointUri))
    {
        throw new Exception("Invalid value used for 'KeyVaultName' property. Please provide a valid key-vault name using the 'OrchardCore:OrchardCore_KeyVault_Azure:KeyVaultName' settings key.");
    }
}
```

I would even go an extra step an more that login into a separate private method using something like this:

```csharp
private Uri GetVaultHost()
{
  var vaultUri = configuration["OrchardCore:OrchardCore_KeyVault_Azure:VaultURI"];
  
  if (!string.IsNullOrWhiteSpace(vaultUri))
  {
      if(!Uri.TryCreate(vaultUri, UriKind.Absolute, out uri))
      {
          throw new Exception("Invalid value used for 'VaultURI' property. Please provide a valid key-vault name using the 'OrchardCore:OrchardCore_KeyVault_Azure:KeyVaultName' settings key.");
      }
  
      return uri;
  } 
  
    var keyVaultName = configuration["OrchardCore:OrchardCore_KeyVault_Azure:KeyVaultName"];
    
    if (string.IsNullOrEmpty(keyVaultName))
    {
        throw new Exception("The 'KeyVaultName' property is no configured. Please configure it by specifying the 'OrchardCore:OrchardCore_KeyVault_Azure:KeyVaultName' settings key.");
    }
    
    if (!Uri.TryCreate($"https://{keyVaultName}.vault.azure.net", UriKind.Absolute, out var host))
    {
        throw new Exception("Invalid value used for 'KeyVaultName' property. Please provide a valid key-vault name using the 'OrchardCore:OrchardCore_KeyVault_Azure:KeyVaultName' settings key.");
    }
    
    return host;
}

```

Reply:
@MikeAlhayek  - The approach using the GetVaultHost() method seems more appropriate. I'm making the necessary changes and will commit them again.